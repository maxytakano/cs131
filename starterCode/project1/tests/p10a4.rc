function : void main() {
	
	int c[5];
	int ca;
	ca = c[1] + c[1]; 	// valid

	int b[5];
	int ba;
	ba = b + b; 		// invalid. incompatible types to + 

	int d[5];
	d[0]++;				// valid
	d[1]--;				// valid
	
	float g[1];
	g++; 				// error. incompatible type to ++
	g--; 				// error. incompatible type to --
	!g;					// error. incompatible type to !
	
}