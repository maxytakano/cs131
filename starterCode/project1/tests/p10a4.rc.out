Error, "/tmp/cs131_testrunner_SyTaux/file.rc", line 9: 
  Incompatible type int[5] to binary operator +, numeric expected.
Error, "/tmp/cs131_testrunner_SyTaux/file.rc", line 16: 
  Incompatible type float[1] to operator ++, equivalent to int, float, or pointer expected.
Error, "/tmp/cs131_testrunner_SyTaux/file.rc", line 17: 
  Incompatible type float[1] to operator --, equivalent to int, float, or pointer expected.
Error, "/tmp/cs131_testrunner_SyTaux/file.rc", line 18: 
  Incompatible type float[1] to unary operator !, equivalent to bool expected.
Compile: failure.
